src/lib/source-handlers/azure/list-projects.ts

```diff
@@ -0,0 +1,113 @@
+import { AzureProjectData as Project } from './types';
+import base64 = require('base-64');
+import Bottleneck from 'bottleneck';
+import * as debugLib from 'debug';
+import { AzureProjectData } from './types';
+import { getAzureToken } from './get-azure-token';
+import { getAzureOrgName } from './get-azure-orgname';
+import * as needle from 'needle';
+
+const debug = debugLib('snyk:azure');
+const azureDefaultUrl = 'https://dev.azure.com';
+
+const limiter = new Bottleneck({
+  maxConcurrent: 1,
+  minTime: 500,
+});
+
+limiter.on('failed', async (error, jobInfo) => {
+  const id = jobInfo.options.id;
+  console.warn(`Job ${id} failed: ${error}`);
+  if (jobInfo.retryCount === 0) {
+    // Here we only retry once
+    console.log(`Retrying job ${id} in 25ms!`);
+    return 25;
+  }
+});
+
+async function fetchAllProjects(
+  azureDefaultUrl: string,
+  OrgName: string,
+  token: string,
+  top: number,
+): Promise<Project[]> {
+  const projectList: Project[] = [];
+  let continuationToken = 0;
+  let hasMorePages = true;
+  while (hasMorePages) {
+    try {
+      const { projects, continuationtoken, entries } = await getProjects(
+        azureDefaultUrl,
+        OrgName,
+        token,
+        top,
+        continuationToken,
+      );
+      if (continuationtoken > 0 && entries > 0) {
+        continuationToken = continuationtoken;
+      } else {
+        hasMorePages = false;
+      }
+      projects.map((project: { name: string; id: string }) => {
+        const { name, id } = project;
+        if (name && id) {
+          projectList.push({ id: project.id, name: project.name });
+        }
+      });
+    } catch (err) {
+      debug('Failed to retrieve project list from Azure Devops.\n' + err);
+      console.log(
```

Comment:
use `debug` only here.

Reply:
there is no need to catch here even let it throw so the error bubbles up and the CLI stops (it is built to handle errors and show them to the use, by catching here you are swallowing it and letting the tool continue). Unless you plan to handle a failed page with a retry remove `try/catch`

Reply:
Done